ush/rocoto/workflow_tasks.py

```diff
@@ -0,0 +1,1065 @@
+
+        return task
+
+    def prep(self):
+
+        suffix = self._base["SUFFIX"]
+        dump_suffix = self._base["DUMP_SUFFIX"]
+        gfs_cyc = self._base["gfs_cyc"]
+        dmpdir = self._base["DMPDIR"]
+        gfs_enkf = True if self.app_config.do_hybvar and 'gfs' in self.app_config.eupd_cdumps else False
+
+        deps = []
+        dep_dict = {'type': 'metatask', 'name': f'{"gdas"}post', 'offset': '-06:00:00'}
+        deps.append(rocoto.add_dependency(dep_dict))
+        data = f'&ROTDIR;/gdas.@Y@m@d/@H/atmos/gdas.t@Hz.atmf009{suffix}'
+        dep_dict = {'type': 'data', 'data': data, 'offset': '-06:00:00'}
+        deps.append(rocoto.add_dependency(dep_dict))
+        data = f'{dmpdir}/{self.cdump}{dump_suffix}.@Y@m@d/@H/{self.cdump}.t@Hz.updated.status.tm00.bufr_d'
+        dep_dict = {'type': 'data', 'data': data}
+        deps.append(rocoto.add_dependency(dep_dict))
+        dependencies = rocoto.create_dependency(dep_condition='and', dep=deps)
+
+        cycledef = self.cdump
+        if self.cdump in ['gfs'] and gfs_enkf and gfs_cyc != 4:
+            cycledef = 'gdas'
+
+        resources = self.get_resource('prep')
+        task = create_wf_task('prep', resources, cdump=self.cdump, envar=self.envars, dependency=dependencies,
+                              cycledef=cycledef)
+
+        return task
+
+    def waveinit(self):
+
+        resources = self.get_resource('waveinit')
+        dependencies = None
+        if self.app_config.mode in ['cycled']:
+            deps = []
+            dep_dict = {'type': 'task', 'name': f'{self.cdump}prep'}
+            deps.append(rocoto.add_dependency(dep_dict))
+            dep_dict = {'type': 'cycleexist', 'condition': 'not', 'offset': '-06:00:00'}
+            deps.append(rocoto.add_dependency(dep_dict))
+            dependencies = rocoto.create_dependency(dep_condition='or', dep=deps)
+
+        task = create_wf_task('waveinit', resources, cdump=self.cdump, envar=self.envars, dependency=dependencies)
+
+        return task
+
+    def waveprep(self):
+
+        deps = []
+        dep_dict = {'type': 'task', 'name': f'{self.cdump}waveinit'}
+        deps.append(rocoto.add_dependency(dep_dict))
+        dependencies = rocoto.create_dependency(dep=deps)
+
+        resources = self.get_resource('waveprep')
+        task = create_wf_task('waveprep', resources, cdump=self.cdump, envar=self.envars, dependency=dependencies)
+
+        return task
+
+    def aerosol_init(self):
+
+        deps = []
+        # Files from current cycle
+        files = ['gfs_ctrl.nc'] + [f'gfs_data.tile{tile}.nc' for tile in range(1, self.n_tiles + 1)]
+        for file in files:
+            data = f'&ROTDIR;/{self.cdump}.@Y@m@d/@H/atmos/INPUT/{file}'
+            dep_dict = {'type': 'data', 'data': data}
+            deps.append(rocoto.add_dependency(dep_dict))
+
+        # Previous cycle
+        dep_dict = {'type': 'cycleexist', 'offset': f'-{self._base["INTERVAL"]}'}
```

Comment:
This should be INTERVAL_GFS, since what aerosol_init cares about is the last forecast cycle.

Reply:
done. please check.